### PR TITLE
Sobel filter for volumes

### PR DESCRIPTION
**Pull request by mcquin**

### CODE REVIEW

File: centrosome/filter.py [hunk 1]
```diff
@@ -13,15 +13,21 @@
...
+import skimage
+import skimage.filters
+
 '''# of points handled in the first pass of the convex hull code'''
 CONVEX_HULL_CHUNKSIZE = 250000
 
 def stretch(image, mask=None):
     '''Normalize an image to make the minimum zero and maximum one
-    
+
```
Comment (0x00b1) [inline centrosome/filter.py]: Would you mind fixing the trailing whitespace issue in a separate pull request?

**Comment by mcquin** [comment]
https://github.com/CellProfiler/centrosome/pull/68

File: centrosome/filter.py (hunk 1 above)
Comment (mcquin) [inline centrosome/filter.py]: https://github.com/CellProfiler/centrosome/pull/68

**Comment by mcquin** [comment]
@0x00b1 trailing whitespace fixed in #68. this branch is updated.

File: centrosome/filter.py [hunk 2]
```diff
@@ -493,20 +496,59 @@ def roberts(image, mask=None):
...
+        return np.sqrt(hsobel(data, mask)**2 + vsobel(data, mask)**2)
+
+    result = np.sqrt(sobel_x(data, mask)**2 + sobel_y(data, mask)**2 + sobel_z(data, mask)**2)
+    result /= np.sqrt(3)
+    return result
+
+
+def sobel_x(data, mask=None):
+    data = skimage.img_as_float(data)
+    kernel = (__u.transpose(__zt) * __v * __u.transpose(__yt)).astype(float) / 16.0
```
Comment (0x00b1) [inline centrosome/filter.py]: `.astype(float) / 16.0`?

Comment (mcquin) [inline centrosome/filter.py]: Normalization. In 2D it is 4.0, so we square it in 3D to get 16.0. Ensures result is in [0, 1].

Comment (0x00b1) [inline centrosome/filter.py]: @mcquin Is `skimage.exposure.rescale_intensity` preferable?

**Pull request closed without merging**